Log which filename caused a parse error on the demo page

If parsing errors, the demo page should log which file caused the error. Useful for when you try to import a folder of files.

These errors currently get shown, but aren't really helpful:

`main.js:57 No supported broker found!`

```js
getBroker @ index.js:31
getActivity @ index.js:9
extractActivities @ index.js:72
async function (async)
extractActivities @ index.js:55
fileReader.onload @ main.js:57
load (async)
createActivityFromPDF @ main.js:56
processFiles @ main.js:51
fileHandler @ main.js:48
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7547
Show 4 more frames
main.js:71 []length: 0__proto__: Array(0)
onvista.js:145 Error while parsing page (onvista RangeError: Invalid time value
    at format (index.js:371)
    at parseData (onvista.js:122)
    at Module.parsePages (onvista.js:142)
    at getActivity (index.js:13)
    at extractActivities (index.js:72)
    at async FileReader.fileReader.onload (main.js:57) (116) [ FILE CONTENTS AS ARRAY ]
parsePages @ onvista.js:145
getActivity @ index.js:13
extractActivities @ index.js:72
async function (async)
extractActivities @ index.js:55
fileReader.onload @ main.js:57
load (async)
createActivityFromPDF @ main.js:56
processFiles @ main.js:51
fileHandler @ main.js:48
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7547
Show 3 more frames
main.js:71 []
```

--- a/demo/index.html
+++ b/demo/index.html
@@ -60,6 +60,13 @@
                         </p>
                       </div>
                     </div>
+                    <div class="level-right">
+                      <div class="level-item">
+                        <p>
+                          <span><small><a href="https://github.com/codeofsumit/tresor-import/blob/master/docs/status_code.md" target="_blank">(Status-Codes)</a></small></span>
+                        </p>
+                      </div>
+                    </div>
                   </div>
                   <p style="margin: .5em 0 0 .3em">
                     <small>Datei: {{ error.file }}</small>

--- a/demo/index.html
+++ b/demo/index.html
@@ -76,7 +76,7 @@
                   </p>
                   <p style="margin-left: .3em">
                     <textarea :id="'content-' + i" style="display: none">{{ error.content }}</textarea>
-                    <small><a href="#" @click="copyContentToClipboard(i)">Inhalt kopieren</a></small>
+                    <small><a href="#" @click="copyContentToClipboard(i)">JSON-Inhalt kopieren</a></small>
                   </p>
                 </div>
               </li>

--- a/demo/index.html
+++ b/demo/index.html
@@ -48,59 +48,83 @@
             </div>
           </div>
           <br>
-          <ul v-cloak v-show="activities.length > 0" class="list">
-            <li class="list-item" v-for="(activity, i) in activities" :key="i">
-              <div v-if="!activity.parserError">
+            <ul v-cloak v-show="errors.length > 0" class="list">
+              <li class="list-item" v-for="(error, i) in errors" :key="i">
                 <div>
                   <div class="level is-marginless is-mobile">
                     <div class="level-left">
                       <div class="level-item">
-                        <p v-if="activity.type">
-                          <span class="tag is-success is-light"
-                            :class="getTypeColor(activity.type)">{{activity.type}}</span>
-                          <span class="tag is-light">{{activity.shares}}x</span>
-                        </p>
-                      </div>
-                    </div>
-                    <div class="level-right">
-                      <div class="level-item has-text-right">
-                        <p v-if="activity.amount !== undefined">
-                          <strong :class="getPriceColor(activity.type)">
-                            <small>{{formatPrice(activity.amount)}}€</small>
-                          </strong>
+                        <p>
+                          <span class="tag is-success is-light is-warning">Status-Code:</span>
+                          <span class="tag is-light">{{ error.status }}</span>
                         </p>
                       </div>
                     </div>
                   </div>
-                  <p style="margin: .5em 0 0">
-                    <strong>
-                      <small v-if="activity.holding">{{getHoldingName(activity.holding)}}</small>
-                      <small v-else>{{activity.company}}</small>
-                    </strong>
+                  <p style="margin: .5em 0 0 .3em">
+                    <small>Datei: {{ error.file }}</small>
                   </p>
-
-                  <p>
-                    <small class="is-size-7">Tax: {{formatPrice(activity.tax)}}€</small>
+                  <p style="margin-left: .3em">
+                    <small>Anzahl Seiten: 1</small>
                   </p>
-                  <p>
-                    <small class="is-size-7">Fee: {{formatPrice(activity.fee)}}€</small>
+                  <p style="margin-left: .3em">
+                    <input :id="'content-' + i" type="text" :value="error.content" style="display: none" />
+                    <small><a href="#" @click="copyContentToClipboard('content-0')">Inhalt kopieren</a></small>
                   </p>
-
-                  <div class="level is-mobile">
-                    <div class="level-left">
-                      <div class="level-item">
-                        <p style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap; max-width: 150px;"
-                          :title="activity.filename">
-                          <small v-if="activity.filename" class="is-size-7">{{activity.filename}}</small>
-                        </p>
-                      </div>
+                </div>
+              </li>
+            </ul>
+            <ul v-cloak v-show="activities.length > 0" class="list">
+            <li class="list-item" v-for="(activity, i) in activities" :key="i">
+              <div>
+                <div class="level is-marginless is-mobile">
+                  <div class="level-left">
+                    <div class="level-item">
+                      <p v-if="activity.type">
+                        <span class="tag is-success is-light"
+                          :class="getTypeColor(activity.type)">{{ activity.type }}</span>
+                        <span class="tag is-light">{{ activity.shares }}x</span>
+                      </p>
                     </div>
-                    <div class="level-right">
-                      <div class="level-item">
-                        <p>
-                          <small class="is-size-7 has-text-grey">{{formatDate(activity.date)}}</small>
-                        </p>
-                      </div>
+                  </div>
+                  <div class="level-right">
+                    <div class="level-item has-text-right">
+                      <p v-if="activity.amount !== undefined">
+                        <strong :class="getPriceColor(activity.type)">
+                          <small>{{ formatPrice(activity.amount) }}€</small>
+                        </strong>
+                      </p>
+                    </div>
+                  </div>
+                </div>
+                <p style="margin: .5em 0 0">
+                  <strong>
+                    <small v-if="activity.holding">{{ getHoldingName(activity.holding) }}</small>
+                    <small v-else>{{ activity.company }}</small>
+                  </strong>
+                </p>
+
+                <p>
+                  <small class="is-size-7">Tax: {{ formatPrice(activity.tax) }}€</small>
+                </p>
+                <p>
+                  <small class="is-size-7">Fee: {{ formatPrice(activity.fee) }}€</small>
+                </p>
+
+                <div class="level is-mobile">
+                  <div class="level-left">
+                    <div class="level-item">
+                      <p style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap; max-width: 150px;"
+                        :title="activity.filename">
+                        <small v-if="activity.filename" class="is-size-7">{{ activity.filename }}</small>
+                      </p>
+                    </div>
+                  </div>
+                  <div class="level-right">
+                    <div class="level-item">
+                      <p>
+                        <small class="is-size-7 has-text-grey">{{ formatDate(activity.date) }}</small>
+                      </p>
                     </div>
                   </div>
                 </div>

--- a/demo/index.html
+++ b/demo/index.html
@@ -68,8 +68,8 @@
                     <small>Anzahl Seiten: 1</small>
                   </p>
                   <p style="margin-left: .3em">
-                    <input :id="'content-' + i" type="text" :value="error.content" style="display: none" />
-                    <small><a href="#" @click="copyContentToClipboard('content-0')">Inhalt kopieren</a></small>
+                    <textarea :id="'content-' + i" style="display: none">{{ error.content }}</textarea>
+                    <small><a href="#" @click="copyContentToClipboard(i)">Inhalt kopieren</a></small>
                   </p>
                 </div>
               </li>

--- a/demo/index.html
+++ b/demo/index.html
@@ -65,7 +65,7 @@
                     <small>Datei: {{ error.file }}</small>
                   </p>
                   <p style="margin-left: .3em">
-                    <small>Anzahl Seiten: 1</small>
+                    <small>Anzahl Seiten: {{ error.content.length }}</small>
                   </p>
                   <p style="margin-left: .3em">
                     <textarea :id="'content-' + i" style="display: none">{{ error.content }}</textarea>

--- a/demo/js/main.js
+++ b/demo/js/main.js
@@ -2,11 +2,9 @@ import formatDistanceToNow from 'date-fns/formatDistanceToNow';
 import parse from 'date-fns/parse';
 import { de } from 'date-fns/locale';
 // Use the webpack version to ensure, that the published version works fine and not only the src/ one.
-import getActivities, {
-  parseActivitiesFromPages,
-} from '../bundle/tresor-import';
+import { parseFile, parseActivitiesFromPages } from '../bundle/tresor-import';
 // To use the published version, uncomment the following line after running: npm run build
-// import getActivities, { parseActivitiesFromPages } from '../../dist/tresor-import';
+// import { parseFile, parseActivitiesFromPages } from '../../dist/tresor-import';
 
 new Vue({
   el: '#app',
@@ -99,12 +97,22 @@ new Vue({
       copyText.style.display = 'none';
     },
     async fileHandler() {
-      const results = await Promise.all(
-        Array.from(this.$refs.myFiles.files).map(getActivities)
-      );
-
       this.clearResults();
-      results.forEach(this.handleParserResults);
+      Array.from(this.$refs.myFiles.files).forEach(file => {
+        parseFile(file).then(parsedFile => {
+          const result = parseActivitiesFromPages(
+            parsedFile.pages,
+            parsedFile.extension
+          );
+
+          result.file = file.name;
+          result.content = parsedFile.pages;
+          result.successful =
+            result.activities !== undefined && result.status === 0;
+
+          this.handleParserResults(result);
+        });
+      });
     },
     clearResults() {
       this.errors = [];

--- a/demo/js/main.js
+++ b/demo/js/main.js
@@ -87,7 +87,7 @@ new Vue({
       });
     },
     copyContentToClipboard(name) {
-      const copyText = document.getElementById(name);
+      const copyText = document.getElementById('content-' + name);
 
       copyText.style.display = 'block';
 

--- a/demo/js/main.js
+++ b/demo/js/main.js
@@ -11,6 +11,7 @@ import getActivities, {
 new Vue({
   el: '#app',
   data: {
+    errors: [],
     activities: [],
     jsonInputActive: false,
     jsonContent: '',
@@ -54,6 +55,7 @@ new Vue({
       }
 
       if (!result.successful) {
+        this.errors.push(result);
         return;
       }
 
@@ -74,19 +76,39 @@ new Vue({
 
       const result = parseActivitiesFromPages(content, this.jsonExtension);
 
+      this.clearResults();
+
       this.handleParserResults({
-        file: 'Json Input',
+        file: 'json.' + this.jsonExtension,
+        content: this.jsonContent,
         activities: result.activities,
         status: result.status,
         successful: result.activities !== undefined && result.status === 0,
       });
+    },
+    copyContentToClipboard(name) {
+      const copyText = document.getElementById(name);
+
+      copyText.style.display = 'block';
+
+      copyText.select();
+      copyText.setSelectionRange(0, 99999);
+
+      document.execCommand('copy');
+
+      copyText.style.display = 'none';
     },
     async fileHandler() {
       const results = await Promise.all(
         Array.from(this.$refs.myFiles.files).map(getActivities)
       );
 
+      this.clearResults();
       results.forEach(this.handleParserResults);
+    },
+    clearResults() {
+      this.errors = [];
+      this.activities = [];
     },
   },
 });

--- a/src/index.js
+++ b/src/index.js
@@ -55,6 +55,65 @@ export const parseActivitiesFromPages = (pages, extension) => {
   };
 };
 
+export const parseFile = file => {
+  return new Promise(resolve => {
+    const extension = file.name.split('.').pop().toLowerCase();
+    const reader = new FileReader();
+
+    reader.onload = async e => {
+      let fileContent, pdfDocument;
+      let pages = [];
+
+      if (extension === 'pdf') {
+        fileContent = new Uint8Array(e.currentTarget.result);
+        pdfDocument = await pdfjs.getDocument(fileContent).promise;
+
+        const loopHelper = Array.from(Array(pdfDocument.numPages)).entries();
+        for (const [pageIndex] of loopHelper) {
+          pages.push(
+            await parsePageToContent(await pdfDocument.getPage(pageIndex + 1))
+          );
+        }
+      } else {
+        pages.push(e.currentTarget.result.trim().split('\n'));
+      }
+
+      resolve({
+        pages,
+        extension,
+      });
+    };
+
+    if (extension === 'pdf') {
+      reader.readAsArrayBuffer(file);
+    } else {
+      reader.readAsText(file);
+    }
+  });
+};
+
+export default file => {
+  return new Promise(resolve => {
+    try {
+      parseFile(file).then(parsedFile => {
+        const result = parseActivitiesFromPages(
+          parsedFile.pages,
+          parsedFile.extension
+        );
+
+        resolve({
+          file: file.name,
+          activities: result.activities,
+          status: result.status,
+          successful: result.activities !== undefined && result.status === 0,
+        });
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  });
+};
+
 const filterResultActivities = result => {
   if (result.activities !== undefined) {
     result.activities = result.activities.filter(
@@ -81,49 +140,4 @@ const parsePageToContent = async page => {
   }
 
   return parsedContent.filter(item => item.length > 0);
-};
-
-export default file => {
-  return new Promise(resolve => {
-    try {
-      const extension = file.name.split('.').pop().toLowerCase();
-      const reader = new FileReader();
-
-      reader.onload = async e => {
-        let fileContent, pdfDocument;
-        let pages = [];
-
-        if (extension === 'pdf') {
-          fileContent = new Uint8Array(e.currentTarget.result);
-          pdfDocument = await pdfjs.getDocument(fileContent).promise;
-
-          const loopHelper = Array.from(Array(pdfDocument.numPages)).entries();
-          for (const [pageIndex] of loopHelper) {
-            pages.push(
-              await parsePageToContent(await pdfDocument.getPage(pageIndex + 1))
-            );
-          }
-        } else {
-          pages.push(e.currentTarget.result.trim().split('\n'));
-        }
-
-        const result = parseActivitiesFromPages(pages, extension);
-
-        resolve({
-          file: file.name,
-          activities: result.activities,
-          status: result.status,
-          successful: result.activities !== undefined && result.status === 0,
-        });
-      };
-
-      if (extension === 'pdf') {
-        reader.readAsArrayBuffer(file);
-      } else {
-        reader.readAsText(file);
-      }
-    } catch (error) {
-      console.error(error);
-    }
-  });
 };

